service/share/discovery.go

```diff
@@ -87,13 +105,26 @@ func (d *discovery) findPeers(ctx context.Context) {
 			for peer := range peers {
 				go d.handlePeerFound(ctx, topic, peer)
 			}
+		case e := <-sub.Out():
+			// listen to disconnect event to remove peer from set and reset backoff time
+			// reset timer in order to restart the discovery, once stored peer is disconnected
+			connStatus := e.(event.EvtPeerConnectednessChanged)
+			if connStatus.Connectedness == network.NotConnected {
+				if d.set.Contains(connStatus.Peer) {
+					d.connector.Reset(connStatus.Peer)
+					d.set.Remove(connStatus.Peer)
+					d.host.ConnManager().UntagPeer(connStatus.Peer, topic)
+					t.Reset(interval * 3)
```

Comment:
Do we test that discovery is actually restarted?

Reply:
Will try. But the problem is that test could take much time to finish.